src/ngraph/ngraph_utils.h

```diff
@@ -262,6 +259,12 @@ std::ostream & container_to_debug_stream(
   return os;
 }
 
+// generates hash for any standard type val, and combines with seed.
```

Comment:
it is referring to cpp standard types (eg: char, int etc).

Reply:
I hate (really!) to be pedantic, but I still don't know what you mean.

Your examples are C++ "integral" types.  But in standard C++, there are lots of other type categories that you might mean instead:
- C++ [Plain Old Data (POD) types](https://stackoverflow.com/a/146589/387887)
- Various type categories listed [here](http://en.cppreference.com/w/cpp/header/type_traits).
- Any type that's provided by the C++ standard library via the `std::` namespace,
- etc.

It may seem like I'm just being a PITA with this question, but it actually matters w.r.t. hash functions.

Reply:
no, good question actually. "basic types" as defined under `Standard specializations for basic types` http://en.cppreference.com/w/cpp/utility/hash under

Reply:
So maybe the comment should say something like: "T must be a type for which `std::hash<T>` is defined."?

I think that says it all.  Especially because people can (and do) add specializations of `std::hash<...>` for their own, user-defined types.

Reply:
:) It's Ashok's utility, I just moved it.

Reply:
> :) It's Ashok's utility, I just moved it.

Oh Matt...

Reply:
👎 to me for abdicating responsibility.